### ISSUE TITLE
VkExportMemoryWin32HandleInfoKHR should extend VkExportMemoryAllocateInfo

### ISSUE THREAD

**Issue by maierfelix** [KhronosGroup/Vulkan-Docs#1095]
According to [here](https://github.com/NVIDIA/cuda-samples/blob/master/Samples/vulkanImageCUDA/vulkanImageCUDA.cu#L1550)

###

[VkExportMemoryWin32HandleInfoKHR](https://www.khronos.org/registry/vulkan/specs/1.1-extensions/man/html/VkExportMemoryWin32HandleInfoKHR.html) extends [VkExportMemoryAllocateInfo](https://www.khronos.org/registry/vulkan/specs/1.1-extensions/man/html/VkExportMemoryAllocateInfo.html), but there seems to be a mismatch in the XML specification:

````xml
<type category="struct" name="VkExportMemoryWin32HandleInfoKHR" structextends="VkMemoryAllocateInfo">
````

should be:
````xml
<type category="struct" name="VkExportMemoryWin32HandleInfoKHR" structextends="VkExportMemoryAllocateInfo">
````

In my code generator I interpret `structextends` as allowing the parent struct's *pNext* member to be filled with the child struct.

I hope I didn't miss something obvious or got lost in the inheritance chain ;)

**Comment by cubanismo** [comment]
Logically, it extends the VkExportMemoryAllocateInfo structure, but apparently that's not how the generators like things, so you're correct.  @maierfelix, Would you like to put together a pull request to fix this?

**Comment by Tobski** [comment]
Just to be clear, the xml is correct, it's the spec that should be updated. The reason for this is that it's all one big logical chain of structs from the parent, and there's no rule in the spec for any ordering in the chain. If the Win32 struct appears in the chain before the base ExternalMemory struct, that's completely fine; implementations should handle this. Your generator is doing the right thing as it stands, the spec is incorrect.

We made this change a while back as it vastly simplifies the handling for a whole bunch of cases, notably how we generate valid usage conditions; but I guess there are some older structs which are still described in the spec as extending one of the other structs in the chain.

**Comment by cubanismo** [comment]
Yes, sorry, I read the proposal backwards.  Given our current object model, it does seem that what is needed is a bit of language clarifying how the struct is used in the main spec.  Similar changes will be needed for VkExportSemaphoreWin32HandleInfoKHR and VkExportFenceWin32HandleInfoKHR for consistency/completeness.

**Comment by stonesthrow** [comment]
@maierfelix would you like to craft a proposal/MR for this?

**Comment by stonesthrow** [comment]
Moving to Khronos internal issue for resolution.

**Comment by oddhack** [comment]
This should be fixed in the 1.2.135 spec update.
